From 4b3a2f8101921303ebe43b8050272dd0eae5fe8c Mon Sep 17 00:00:00 2001
From: Ondrej Jirman <megi@xff.cz>
Date: Sun, 2 Aug 2026 13:37:49 +0200
Subject: [PATCH] ARM: dts: allwinner: sun8i-a83t: Add UART DMA channels

All five UARTs are hooked up to the DMA controller, but the channels
were never described. 8250_dw sets up->dma unconditionally whenever the
port has a valid fifo-size, so the request fails and the port silently
falls back to PIO:

  dw-apb-uart 1c28400.serial: failed to request DMA

Add the DMA channels. The DRQ assignments are the same as on the rest
of the family (A31, H3, A64), which already agree with this SoC on
every port described so far (SPDIF, I2S0-2).

Signed-off-by: Ondrej Jirman <megi@xff.cz>
---
 arch/arm/boot/dts/allwinner/sun8i-a83t.dtsi | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/arch/arm/boot/dts/allwinner/sun8i-a83t.dtsi b/arch/arm/boot/dts/allwinner/sun8i-a83t.dtsi
index b3f4412cffe2..5a7a26ed97b2 100644
--- a/arch/arm/boot/dts/allwinner/sun8i-a83t.dtsi
+++ b/arch/arm/boot/dts/allwinner/sun8i-a83t.dtsi
@@ -960,6 +960,8 @@ uart0: serial@1c28000 {
 			fifo-size = <64>;
 			clocks = <&ccu CLK_BUS_UART0>;
 			resets = <&ccu RST_BUS_UART0>;
+			dmas = <&dma 6>, <&dma 6>;
+			dma-names = "tx", "rx";
 			status = "disabled";
 		};
 
@@ -972,6 +974,8 @@ uart1: serial@1c28400 {
 			fifo-size = <64>;
 			clocks = <&ccu CLK_BUS_UART1>;
 			resets = <&ccu RST_BUS_UART1>;
+			dmas = <&dma 7>, <&dma 7>;
+			dma-names = "tx", "rx";
 			status = "disabled";
 		};
 
@@ -984,6 +988,8 @@ uart2: serial@1c28800 {
 			fifo-size = <64>;
 			clocks = <&ccu CLK_BUS_UART2>;
 			resets = <&ccu RST_BUS_UART2>;
+			dmas = <&dma 8>, <&dma 8>;
+			dma-names = "tx", "rx";
 			status = "disabled";
 		};
 
@@ -996,6 +1002,8 @@ uart3: serial@1c28c00 {
 			fifo-size = <64>;
 			clocks = <&ccu CLK_BUS_UART3>;
 			resets = <&ccu RST_BUS_UART3>;
+			dmas = <&dma 9>, <&dma 9>;
+			dma-names = "tx", "rx";
 			status = "disabled";
 		};
 
@@ -1008,6 +1016,8 @@ uart4: serial@1c29000 {
 			fifo-size = <64>;
 			clocks = <&ccu CLK_BUS_UART4>;
 			resets = <&ccu RST_BUS_UART4>;
+			dmas = <&dma 10>, <&dma 10>;
+			dma-names = "tx", "rx";
 			status = "disabled";
 		};
 
-- 
2.43.0

